tests/network/localnet/conftest.py

```diff
@@ -57,62 +65,196 @@ def vlan_id(vlan_index_number: Generator[int]) -> int:
 ) -> Generator[netattachdef.NetworkAttachmentDefinition]:
-    with localnet_nad(namespace=namespace_localnet_1.name, name="test-localnet-nad1", vlan_id=vlan_id) as nad:
+    with localnet_nad(
+        namespace=namespace_localnet_1.name,
+        name="test-localnet-nad1",
+        vlan_id=vlan_id,
+        network_name=LOCALNET_BR_EX_NETWORK,
+    ) as nad:
         yield nad
 
 
 @pytest.fixture(scope="module")
 def nad_localnet_2(
     namespace_localnet_2: Namespace, vlan_id: int
 ) -> Generator[netattachdef.NetworkAttachmentDefinition]:
-    with localnet_nad(namespace=namespace_localnet_2.name, name="test-localnet-nad2", vlan_id=vlan_id) as nad:
+    with localnet_nad(
+        namespace=namespace_localnet_2.name,
+        name="test-localnet-nad2",
+        vlan_id=vlan_id,
+        network_name=LOCALNET_BR_EX_NETWORK,
+    ) as nad:
         yield nad
 
 
 @pytest.fixture(scope="module")
-def vm_localnet_1(nad_localnet_1: netattachdef.NetworkAttachmentDefinition) -> Generator[BaseVirtualMachine]:
+def ipv4_localnet_address_pool() -> Generator[str]:
+    return (f"10.0.0.{host_value}/24" for host_value in range(1, 254))
+
+
+@pytest.fixture(scope="module")
+def vm_localnet_1(
+    ipv4_localnet_address_pool: Generator[str], nad_localnet_1: netattachdef.NetworkAttachmentDefinition
+) -> Generator[BaseVirtualMachine]:
     with localnet_vm(
-        namespace=nad_localnet_1.namespace, name="test-vm1", network=nad_localnet_1.name, cidr="10.0.0.1/24"
+        namespace=nad_localnet_1.namespace,
+        name="test-vm1",
+        nad_name=nad_localnet_1.name,
+        cidr=next(ipv4_localnet_address_pool),
+        spec_logical_network=LOCALNET_BR_EX_NETWORK,
     ) as vm:
         yield vm
 
 
 @pytest.fixture(scope="module")
-def vm_localnet_2(nad_localnet_2: netattachdef.NetworkAttachmentDefinition) -> Generator[BaseVirtualMachine]:
+def vm_localnet_2(
+    ipv4_localnet_address_pool: Generator[str], nad_localnet_2: netattachdef.NetworkAttachmentDefinition
+) -> Generator[BaseVirtualMachine]:
     with localnet_vm(
-        namespace=nad_localnet_2.namespace, name="test-vm2", network=nad_localnet_2.name, cidr="10.0.0.2/24"
+        namespace=nad_localnet_2.namespace,
+        name="test-vm2",
+        nad_name=nad_localnet_2.name,
+        cidr=next(ipv4_localnet_address_pool),
+        spec_logical_network=LOCALNET_BR_EX_NETWORK,
     ) as vm:
         yield vm
 
 
 @pytest.fixture(scope="module")
-def vms_localnet(
+def server_client_localnet_vms(
     vm_localnet_1: BaseVirtualMachine, vm_localnet_2: BaseVirtualMachine
 ) -> Generator[tuple[BaseVirtualMachine, BaseVirtualMachine]]:
-    vms = (vm_localnet_1, vm_localnet_2)
-    for vm in vms:
-        vm.start()
-    for vm in vms:
-        vm.wait_for_ready_status(status=True)
-        vm.vmi.wait_for_condition(condition=vm.Condition.Type.AGENT_CONNECTED, status=vm.Condition.Status.TRUE)
-    yield vms
+    server_vm, client_vm = run_vms(vms=[vm_localnet_1, vm_localnet_2])
```

Comment:
any reason why to save `server_vm, client_vm`? call `run_vms` and then yield `vm_localnet_1, vm_localnet_2`

Reply:
I think it is more readable this way - it is clear what is the server and what is the client.

Reply:
Originally, `run_vms` returned a variable length tuple, so it was not possible to return it as is due to the typing limitations.
If the fixture will return a list, that will allow direct return.

However, I prefer we have the function helper return a tuple (feels more correct to me) and the fixture returning only two members as it shows up now.
This is not converging, so I guess we can handle it as a follow up.